hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/message/ResponseSegment.java

```diff
@@ -69,7 +71,13 @@ public String constructMSH(HL7Message messageObj) {
 
 		// When constructing responses directly in HNS ESB (as opposed to something returned from a downstream system)
 		// the Message Type will always be ACK
-		sb.append(Util.ACK);
+		// For HNETDTTN it will be NMR
+		if (StringUtils.isNotBlank(messageObj.getReceivingApplication())
+				&& messageObj.getReceivingApplication().equalsIgnoreCase(Util.HNETDTTN)) {
```

Comment:
If you just do StringUtils.equalsIgnoreCase() it will handle the case where receivingApplication is null/blank/empty.